components/patient-data/rest/src/test/java/org/phenotips/data/rest/internal/DefaultPatientByExternalIdResourceImplTest.java

```diff
@@ -0,0 +1,345 @@
+        this.component = (DefaultPatientByExternalIdResourceImpl) this.mocker.getComponentUnderTest();
+        this.logger = this.mocker.getMockedLogger();
+        Provider<XWikiContext> provider = this.mocker.getInstance(XWikiContext.TYPE_PROVIDER);
+        this.context = provider.get();
+        ReflectionUtils.setFieldValue(this.component, "uriInfo", this.uriInfo);
+
+        doReturn(this.uriBuilder).when(this.uriInfo).getBaseUriBuilder();
+        when(this.patient.getId()).thenReturn(this.id);
+        when(this.users.getCurrentUser()).thenReturn(null);
+        when(this.patient.getDocument()).thenReturn(null);
+    }
+
+    @Test
+    public void getPatientWithNoAccessReturnsForbiddenCode() throws ComponentLookupException, XWikiRestException
+    {
+        when(this.repository.getPatientByExternalId(this.eid)).thenReturn(this.patient);
+        when(this.access.hasAccess(Right.DELETE, null, null)).thenReturn(false);
+
+        Response response = this.component.getPatient(this.eid);
+        verify(this.logger).debug("Retrieving patient record with external ID [{}] via REST", this.eid);
+        verify(this.logger).debug("View access denied to user [{}] on patient record [{}]", null, this.patient.getId());
+
+        assertEquals(Response.Status.FORBIDDEN.getStatusCode(), response.getStatus());
+    }
+
+    @Test
+    public void getPatientPerformsCorrectly() throws ComponentLookupException, XWikiRestException
+    {
+        when(this.repository.getPatientByExternalId(this.eid)).thenReturn(this.patient);
+        when(this.access.hasAccess(Right.VIEW, null, null)).thenReturn(true);
+
+        JSONObject jsonObject = new JSONObject();
+        when(this.patient.toJSON()).thenReturn(jsonObject);
+        when(this.uriInfo.getBaseUriBuilder()).thenReturn(this.uriBuilder);
+        when(this.uriBuilder.path(PatientResource.class)).
+                thenReturn(uriBuilder);
+        when(this.uriBuilder.build(this.patient.getId())).thenReturn(this.uri);
+
+        Response response = this.component.getPatient(this.eid);
+        verify(this.logger).debug("Retrieving patient record with external ID [{}] via REST", this.eid);
+
+        JSONObject links = new JSONObject().accumulate("rel", Relations.SELF).accumulate("href", "uri");
+        JSONObject json = new JSONObject().accumulate("links", links);
+
+        assertEquals(json, response.getEntity());
+        assertEquals(Response.Status.OK.getStatusCode(), response.getStatus());
+    }
+
+    @Test
+    public void getPatientNotFoundChecksForMultipleRecords() throws ComponentLookupException, QueryException, XWikiRestException
+    {
+        Query query = mock(DefaultQuery.class);
+        when(this.repository.getPatientByExternalId(this.eid)).thenReturn(null);
+        when(this.qm.createQuery(Matchers.anyString(), Matchers.anyString())).thenReturn(query);
+        when(query.execute()).thenReturn(new ArrayList<Object>());
```

Comment:
Try testing for the condition where the query returns a list of size > 1 as well